Support for FBlocks, Refactoring in MC7 parsing and Bug fixes

StefanHasensperling opened the pull request:
Added support for F-Block (Safety PLC programming) languages and block attributes

Refactored Interface declaration Parsing. The "Parameter.cs" is now much easier to follow and has a lot of bugs fixed. Also a lot of Documentation was added. The Interface Parsing now supports the following points:

-Start Values or Initial Values are now correctly parsed and handled form the MC7 code. All/most datatypes are supported, DateAndTime, Strings.... 
This point caused me a lot of grief and it really shows what a mess the S7 really is. 
-Multi dimensional Arrays are now correctly handled
-Structures with more than 255 members are now correctly handled.
-Current values are now parsed correctly 

I also did many tests and wrote some Unit tests.

StefanHasensperling commented:
There are some conflicts in  public enum PLCLanguage, but i dont know how to resolve this

jogibear9988 commented:
you need to merge master to your branch, then solve conflicts, then push

StefanHasensperling commented:
OK, the conflict is resolved. 
Tomorrow i will look into the failing Unit tests. Anyway i wanted to update them some time this week....

Also if PLCLanguage 7 is CFC, then i think 8 would be probably SFC?

jogibear9988 commented:
Your pull req has still errors... you removed my tia v13 and tia v14 csprojects, see https://github.com/dotnetprojects/DotNetSiemensPLCToolBoxLibrary/pull/80/files?w=1

jogibear9988 commented:
if you're not able to fix, tell me, i'll can include your changes manually! :-)

StefanHasensperling commented:
I will check this weekend. I think i have tho check the Start value parsing, because i found an block where i tested, and the Startvalues are incorrectly parse...

StefanHasensperling commented:
If you have time, you can include the changes, and i will check the start value issue later

StefanHasensperling commented:
Sorry for the delay, I was very busy latley, but i think next week i will have time finish the push req. 
I already have made some unit tests and documentation regarding Block interface parsing.

StefanHasensperling commented:
OK, I finally got around to do tests on the MC7 Interface parsing. I fixed a lot of Bugs and the "Parameter.cs" is now parsing all my Testcases correctly. I added unit tests, that include realy complex interface declarations with complex Initial values to be parsed, especially DB3003. 

I also put together an document with all information i have about the MC7 Block format. 
Jochen, maybe you could add some information about the Code block format, mainley the actual MC7 executable code format.

jogibear9988 commented on .gitignore:
```diff
@@ -2,8 +2,8 @@
-compiled/**/*
+compiled/*.pdb
+#compiled/**/*
 *.vshost.exe
```
can you remove this change??

jogibear9988 commented:
why have you removed the two tia projects?

StefanHasensperling commented:
Unintentional. I will look into it next week.

On Sep 16, 2017 20:48, "Jochen Kühner" <notifications@github.com> wrote:

> why have you removed the two tia projects?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/dotnetprojects/DotNetSiemensPLCToolBoxLibrary/pull/80#issuecomment-330002023>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AIyYeZDG7CXynHWcgHbI9zSUIQAu4Eodks5sjF5pgaJpZM4PAzgk>
> .
>

StefanHasensperling commented:
I am still having trouble with the Pull request, i will remove it, and make an new one
